File: VMS_Web/VMS_Web/ClientApp/src/models/geofence.ts
import GeoCoordinate from "./geoCoordinate";

export default class Geofence {
  public id: number|undefined;
  public companyId: number;
  public name: string;
  public coords: GeoCoordinate[];
  public restrictionId: number;
  public color: string;

  constructor(id: number|undefined, companyId: number, name: string, coords: GeoCoordinate[],
              restrictionId: number, color: string) {
    this.id = id;
    this.companyId = companyId;
    this.name = name;
    this.coords = coords;
    this.restrictionId = restrictionId;
    this.color = color;
  }
}

File: VMS_Backend/VMS_Web/Data/DatabaseModels/Geofence.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VMS_Web.Data.DatabaseModels
{
    [Table("geofence")]
    public class Geofence
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        
        [Column("name")]
        [MaxLength(50)]
        public string Name { get; set; }
        
        [Column("company_id")] 
        public int CompanyId { get; set; }
        [ForeignKey("CompanyId")]
        public Company Company { get; set; } 
        
        [Column("coords")]
        public string Coords { get; set; }
        
        [Column("restriction_id")]
        public byte RestrictionId { get; set; }

        [Column("color")]
        [MaxLength(20)]
        public string Color { get; set; }
    }
}